src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -892,14 +949,64 @@ public DescriptorImpl getDescriptor() {
     private static class GroupDetailsImpl extends GroupDetails {
 
         private String name;
+        private final Set<String> members;
 
         public GroupDetailsImpl(String name) {
             this.name = name;
+            this.members = null;
+        }
+
+        public GroupDetailsImpl(String name, Set<String> members) {
+            this.name = name;
+            this.members = members;
         }
 
         public String getName() {
             return name;
         }
+
+        @Override
+        public Set<String> getMembers() {
+            return members;
+        }
+    }
+
+    private static class LdapGroupDetailsMapper implements LdapEntryMapper {
+        @Override
+        public Object mapAttributes(String dn, Attributes attributes) throws NamingException {
+            String cn = fixGroupname(String.valueOf(attributes.get("cn").get()));
+            NamingEnumeration<?> enumeration;
+            boolean expectingUidInsteadOfDn = false;
+            if (attributes.get("member") != null) {
+                enumeration = attributes.get("member").getAll();
+            } else if (attributes.get("uniqueMember") != null) {
+                enumeration = attributes.get("uniqueMember").getAll();
+            } else if (attributes.get("memberUid") != null) {
+                enumeration = attributes.get("memberUid").getAll();
+                expectingUidInsteadOfDn = true;
+            } else {
+                return new GroupDetailsImpl(cn);
+            }
+
+            Set<String> members = new TreeSet<>();
+            while (enumeration.hasMore()) {
+                String memberDn = String.valueOf(enumeration.next());
+                if (expectingUidInsteadOfDn) {
+                    members.add(fixUsername(memberDn));
+                } else {
+                    try {
+                        LdapName memberName = new LdapName(memberDn);
+                        Attributes memberAttributes = memberName.getRdn(memberName.size() - 1).toAttributes();
+                        if (memberAttributes.get("cn") != null) {
```

Comment:
If we want to use the `Display Name LDAP attribute` option set in the LDAP plugin configuration we would need to make an extra query to get all of the attributes for the user. Is the current code an acceptable trade-off for performance, or is it worth the extra query to get the configured display name?